### PR TITLE
casync build: caidx filename is canonical representation of build

### PR DESCRIPTION
**Pull request by jnewb1**
`nightly.caidx` -> `0.9.7-9c8aedf-release.caidx`

then api can point

`/channels/nightly`
to
`https://commadist.azureedge.net/openpilot-releases/0.9.7-9c8aedf-release.caidx`

and `/channels/master`
to
`https://commadist.azureedge.net/openpilot-releases/0.9.7-9c8aedf-debug.caidx`

### CODE REVIEW

File: system/version.py [hunk 1]
```diff
@@ -69,12 +69,17 @@ class OpenpilotMetadata:
...
+  build_style: str
   is_dirty: bool  # whether there are local changes
 
   @property
   def short_version(self) -> str:
     return self.version.split('-')[0]
 
+  @property
+  def short_commit(self) -> str:
+    return self.git_commit[:6]
```
Comment (incognitojam) [inline system/version.py:81]: 6 characters might not be enough to unambiguously identify commits; `git log --abbrev-commit` uses 9 chars in openpilot

Comment (jnewb1) [reply system/version.py]: well it should be a unique identifier so lets do 9

File: system/version.py [hunk 2]
```diff
@@ -103,6 +108,9 @@ def tested_channel(self) -> bool:
   def release_channel(self) -> bool:
     return self.channel in RELEASE_BRANCHES
 
+  def canonical(self) -> str:
+    return f"{self.openpilot.version}-{self.openpilot.short_commit}-{self.openpilot.build_style}"
```
Comment (adeebshihadeh) [inline system/version.py:112]: I was thinking we use the full commit hash here

Comment (jnewb1) [reply system/version.py]: `0.9.7-f24ddc5f507c2cc3f0176b4ec176b7ef953036f8-release.caidx`

done

**Pull request merged**